src/examples/signal_generator.cpp

```diff
@@ -0,0 +1,152 @@
+/**
+ * @file signal_generator.cpp
+ * @author Luciana Moreno Borges (youlumoreno.borges@gmail.com)
+ * @brief Configure the components, connect them and start the state machine
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <memory>
+#include <QApplication>
+#include <QTimer>
+#include <hyro/core/StateMachine.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+#include <hyro/widgets/WidgetCollection.h>
+#include <hyro/utils/StateMachineSpinner.h>
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/DigitalConverterComponent.h>
+
+using namespace hyro;
+/**
+ * @brief Main loop
+ * 
+ * @param argc 
+ * @param argv 
+ * @return int 
+ */
+
+void dynamic_loop(){
+  
+  /* Initialize dynamic properties with a default value */
+  float amplitude = 0.0f;
+  float frequency = 0.0f;
+  bool cosine = true;
+
+  char ch;
+
+  do{
+
+    DynamicPropertyAccess dynamic_property_access("/signal_generator"_uri);
+    std::cout << "Get dynamic properties..." << std::endl;
+
+    /* Read dynamic properties values */
+    dynamic_property_access.get<float>("amplitude", amplitude);
+    dynamic_property_access.get<float>("frequency", frequency);
+    dynamic_property_access.get<bool>("cosine", cosine);
+    std::cout << "Current amplitude: " << amplitude << std::endl << "Current frequency: " << frequency << std::endl;
+
+    /* Get user inputs for dynamic properties */
+    std::string amp_str, freq_str, cosine_str;
+    std::cout << "Please inform the amplitude: ";
+    std::cin >> amp_str;
+    std::cout << "Please inform the frequency: ";
+    std::cin >> freq_str;
+    std::cout << "Please inform the cosine: ";
+    std::cin >> cosine_str;
+
+    amplitude = std::stof(amp_str);
+    frequency = std::stof(freq_str);
+    cosine = std::stof(cosine_str);
```

Comment:
`std::stof` converts string to float. Here you are converting string to `bool`.

You can change the message to:

```c++
std::cout << "Please inform 0 for sine and 1 for cosine:" << std;
```
You can use `std::stoi` to convert string to integer and then do a cast to `bool`.